submissions/yaripey/frogger/app.js

```diff
@@ -0,0 +1,124 @@
+// Tile size
+const tile = {
+    width: 101,
+    height: 85
+}
+
+// Enemies our player must avoid
+const Enemy = function (x, y) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+    this.x = x;
+    this.y = y;
+    this.changeSpeed();
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.changeSpeed = function () {
+    this.speed = Math.random() * (400 - 100) + 100;
+}
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+    if (this.x > tile.width * 5) {
+        this.changeSpeed()
+        this.x = -tile.width * 1
+    }
+    this.x += this.speed * dt;
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+const Player = function (x, y) {
+    this.initialX = x;
+    this.initialY = y;
+    this.x = x;
+    this.y = y;
+    this.stepX = tile.width;
+    this.stepY = tile.height;
+
+    this.sprite = "images/char-boy.png";
```

Comment:
Definition of this.x, this.y, this.sprite - in 'Player' also can be moved to the base 'Character' class...